File: 02-Constructors/Square.java
class Square {
    int side;

    Square(int side) {
        this.side = side;
    }

    Square(Square s) {
        this.side = s.side;
    }

    void displaySquare() {
        System.out.println("Square area: " + this.side*this.side);
    }

    public static void main(String[] args) {
        Square s1 = new Square(4);
        Square s2 = new Square(s1);

        s2.displaySquare();
    }
}

File: 02-Constructors/Square.cpp
#include <iostream>
#include <string>

using namespace std;

class Square {
public:
    int side;

    Square(int side) {
        this->side = side;
    }

    Square(const Square &s) {
        this->side = s.side;
    }

    void displayArea() const {
        cout << "Square area: " << this->side * this->side << endl;
    }
};

int main() {
    Square s1(4);

    Square s2 = s1;

    s2.displayArea(); 

    return 0;
}
